### PR TITLE
added trap that summons spiders as well as the corresponding spell

### PR DESCRIPTION
**Pull request by Lady-Aranae**
#### Summary
Content "Added a trap that summons spiders as well as the required spell"

#### Purpose of change
Trapdoor spiders didn't feel like actual Trappers. Wanted to change that 

#### Describe the solution
Added trap with the ID "trapdoor spiderden" that spawns hostile spiders and an attached classless spell 

#### Describe alternatives you've considered
Considered changing how trapdoor spiders move and fight

#### Testing
Spawned trap, activated trap, checked error log for any issues

### CODE REVIEW

**Comment by actual-nh** [comment]
Ping: @Venera3?

**Comment by Maleclypse** [comment]
Can you add a place where this trap occurs?   Maybe in whatever map extra that trapdoor spiders spawn in?  Or a basement?  I can approve once the trap is somewhere in the game.

**Comment by Venera3** [comment]
You could well include it in vanilla, potentially as a random forest map extra - including it in the existing web spider den would be nicer, but it would involve c++ since those are hardcoded.

File: data/mods/Magiclysm/Spells/classless.json [hunk 1]
```diff
@@ -267,5 +267,32 @@
     "base_casting_time": 200,
     "base_energy_cost": 250,
     "min_duration": 180000
+  },
+  {
+    "id": "Summon_trapspider",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/classless.json:272]:
```suggestion
    "id": "summon_trapspider",
```

File: data/mods/Magiclysm/Spells/classless.json [hunk 2]
```diff
@@ -267,5 +267,32 @@
...
+    "type": "SPELL",
+    "name": "summon trapdoor spider" ,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/classless.json:274]:
```suggestion
    "name": "Summon Trapdoor Spider" ,
```

File: data/mods/Magiclysm/traps.json [hunk 3]
```diff
@@ -25,5 +25,19 @@
...
+	  "id": "tr_spiderden",
+	  "name": "Trapdoor spiderden",
+	  "color": "green",
+	  "symbol": "#",
+	  "visibility": 99,
+	  "avoidance": 99,
+	  "difficulty": 99,
+	  "trap_radius": 4,
+	  "action": "spell",
+	  "spell_data": { "id": "Summon_trapspider" },
```
Comment (KorGgenT) [inline data/mods/Magiclysm/traps.json:40]:
```suggestion
	  "spell_data": { "id": "summon_trapspider" },
```

**Review by KorGgenT — CHANGES_REQUESTED** [review]
The concept looks good, but like maleclypse said, maybe spawn this trap somewhere? also you need to lint your json.

**Comment by Lady-Aranae** [comment]
Hi, I Added the map extra and linted the json. do I make a new pr or are those changes considered part of this pr?

**Comment by actual-nh** [comment]
> Hi, I Added the map extra and linted the json. do I make a new pr or are those changes considered part of this pr?

No, just push those to your Github Lady-Aranae:Trapdoor-spiders branch, and also accept the changes that KorGgenT suggested.

**Comment by KorGgenT** [comment]
sorry about the delay, we've got a json validation error here.

File: data/mods/Magiclysm/Spells/classless.json [hunk 4]
```diff
@@ -267,5 +267,32 @@
...
+    "shape": "blast",
+    "valid_targets": [ "ground" ],
+    "flags": [ "PERMANENT", "HOSTILE_SUMMON"],
+    "max_level": 15,
+	"min_aoe": 2,
+	"max_aoe": 3,
+    "min_damage": 1,
+    "max_damage": 1,
+	"min_range": 1,
+	"max_range": 3,
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/classless.json:287]:
```suggestion
	  "min_aoe": 2,
	  "max_aoe": 3,
    "min_damage": 1,
    "max_damage": 1,
	  "min_range": 1,
	  "max_range": 3,
```

File: data/mods/Magiclysm/Spells/classless.json [hunk 5]
```diff
@@ -267,5 +267,32 @@
...
+    "min_duration": 30000,
+    "max_duration": 120000,
+    "duration_increment": 6000,
+    "spell_class": "NONE",
+    "base_casting_time": 400,
+    "base_energy_cost": 600,
+    "energy_source": "MANA",
+    "difficulty": 6
+
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/classless.json:296]:
```suggestion
```

File: data/mods/Magiclysm/traps.json [hunk 6]
```diff
@@ -25,5 +25,21 @@
...
+  "symbol": "#",
+  "visibility": 99,
+  "avoidance": 99,
+  "difficulty": 99,
+  "trap_radius": 4,
+  "action": "spell",
+  "spell_data": {
+    "id": "Summon_trapspider"
+  },
+  "always_invisible": true
```
Comment (actual-nh) [inline data/mods/Magiclysm/traps.json:43]:
```suggestion
  {
    "type": "trap",
    "id": "tr_spiderden",
    "name": "Trapdoor spiderden",
    "color": "green",
    "symbol": "#",
    "visibility": 99,
    "avoidance": 99,
    "difficulty": 99,
    "trap_radius": 4,
    "action": "spell",
    "spell_data": {
      "id": "Summon_trapspider"
    },
    "always_invisible": true
```

**Review by actual-nh — COMMENTED** [review]
This may not fix all of the JSON errors, but it should help.

File: data/mods/Magiclysm/Spells/classless.json [hunk 7]
```diff
@@ -267,5 +267,32 @@
     "base_casting_time": 200,
     "base_energy_cost": 250,
     "min_duration": 180000
+  },
+  {
+    "id": "summon_trapspider",
+    "type": "SPELL",
+    "name": "Summon Trapdoor Spider" ,
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/classless.json:274]:
```suggestion
    "name": "Summon Trapdoor Spider",
```

File: data/mods/Magiclysm/Spells/classless.json [hunk 8]
```diff
@@ -267,5 +267,32 @@
...
+    "description": "this spell summons hostile trapdoor spiders from the earth",
+    "effect": "summon",
+    "effect_str": "mon_spider_trapdoor_giant",
+    "shape": "blast",
+    "valid_targets": [ "ground" ],
+    "flags": [ "PERMANENT", "HOSTILE_SUMMON"],
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/classless.json:280]:
```suggestion
    "flags": [ "PERMANENT", "HOSTILE_SUMMON" ],
```

**Comment by Lady-Aranae** [comment]
All the suggested changes should be committed now

**Comment by KorGgenT** [comment]
you need to run magiclysm/traps.json through the linter. http://dev.narc.ro/cataclysm/format.html

File: data/mods/Magiclysm/traps.json [hunk 9]
```diff
@@ -25,5 +25,19 @@
     "action": "spell",
     "spell_data": { "id": "forge_trap" },
     "benign": false
+  },
+  {
+    "type": "trap",
+    "id": "tr_spiderden",
```
Comment (Maleclypse) [inline data/mods/Magiclysm/traps.json:31]:
```suggestion
    "id": "tr_spider",
```

**Pull request closed without merging**